doc/server_side_auth.md

```diff
@@ -12,6 +12,13 @@ The authentication context is structured as a multi-map of key-value pairs - the
 
 The contents of the *auth properties* are populated by an *auth interceptor*. The interceptor also chooses which property key will act as the peer identity (e.g. for client certificate authentication this property will be `"x509_common_name"` or `"x509_subject_alternative_name"`).
 
+Note that AuthContext is not modifiable, unless AuthMetadataProcessor is used([reference](https://github.com/grpc/grpc/blob/master/include/grpcpp/impl/codegen/security/auth_context.h#L89)). 
```

Comment:
While you're in this doc, the note on line 5 is no longer correct.  Please consult @yihuazhang for better wording here, but we should instead talk about the security level agreement that has to be in place between the call creds and the security connector.